source/common/common/matchers.cc

```diff
@@ -130,13 +130,30 @@ PathMatcherConstSharedPtr PathMatcher::createExact(const std::string& exact, boo
   return std::make_shared<const PathMatcher>(matcher);
 }
 
-PathMatcherConstSharedPtr PathMatcher::createPrefix(const std::string& prefix, bool ignore_case) {
+namespace {
+PathMatcherConstSharedPtr createPrefixPathMatcher(const std::string& prefix, bool ignore_case) {
   envoy::type::matcher::v3::StringMatcher matcher;
   matcher.set_prefix(prefix);
   matcher.set_ignore_case(ignore_case);
   return std::make_shared<const PathMatcher>(matcher);
 }
 
+} // namespace
+
+PathMatcherConstSharedPtr PathMatcher::createPrefix(const std::string& prefix, bool ignore_case) {
+  if (prefix.empty()) {
```

Comment:
nit: add comments explaining why this is added this way (preventing future "optimizations" to remove redundant code).